### DISCUSSION TITLE
When should I use an atomFamily over atoms in atoms?

### DISCUSSION THREAD

**Discussion by masad-frost** [pmndrs/jotai#1349] [Q&A]
Is there ever a situation where I should use an atomFamily? The only thing I can think of is the ergonomics of implicitly creating an atom if you don't already have one, is that it? Feels more like a footgun than a feature, anyway.

I'm happy to update https://jotai.org/docs/utils/atom-family and https://jotai.org/docs/guides/atoms-in-atom based on the answer.

**Comment by dai-shi** [comment]
If you already have a "key" in your data model, I think `atomFamily` makes more sense.
atoms-in-atom would only make sense if an "atom" itself can be a key.
These two are not mutually exclusive. (Even if you have a "key", you could use an "atom" as a key.)

Hope it helps and you are very welcome to work on docs! (And, I'm happy to help.)
